tests/plugins/test_ckan.py

```diff
@@ -0,0 +1,585 @@
+            "data/ckan_mock_responses/package_show.json",
+        )
+
+        _mock_json_call(
+            f"{self.base_url}/api/3/action/datastore_search?resource_id=_table_metadata&{filter_qs}",
+            "data/ckan_mock_responses/datastore_search_table_metadata_02.json",
+        )
+
+        _mock_json_call(
+            f"{self.base_url}/api/3/action/datastore_create",
+            "data/ckan_mock_responses/doesnt_matter.json",
+            method=responses.POST,
+        )
+
+        _mock_json_call(
+            f"{self.base_url}/api/3/action/datastore_upsert",
+            "data/ckan_mock_responses/doesnt_matter.json",
+            method=responses.POST,
+        )
+
+    @responses.activate
+    def test_write_resource(self):
+        package = Package("data/storage/types.json")
+        package.resources[0].name = self.target_resource_id
+
+        package.resources[0].to_ckan(
+            base_url=self.base_url, dataset_id=self.dataset_id, api_key="env:CKAN_API_KEY"
+        )
+        assert responses.assert_call_count(
+            f"{self.base_url}/api/3/action/datastore_upsert", 1
+        )
+
+    @responses.activate
+    def test_write_package(self):
+        package = Package("data/storage/types.json")
+        package.resources[0].name = self.target_resource_id
+
+        package.resources[0].to_ckan(
+            base_url=self.base_url, dataset_id=self.dataset_id, api_key="env:CKAN_API_KEY"
+        )
+        assert responses.assert_call_count(
+            f"{self.base_url}/api/3/action/datastore_upsert", 1
+        )
+
+    @responses.activate
+    def test_write_types(self):
+        package = Package("data/storage/types.json")
+        package.resources[0].name = self.target_resource_id
+
+        package.resources[0].to_ckan(
+            base_url=self.base_url, dataset_id=self.dataset_id, api_key="env:CKAN_API_KEY"
+        )
+
+        # metdadata
+        assert json.loads(responses.calls[2].request.body) == {
+            "fields": [
+                {"id": "any", "type": "text"},  # type fallback
+                {"id": "array", "type": "text[]"},
+                {"id": "boolean", "type": "bool"},
+                {"id": "date", "type": "text"},  # type downgrade
+                {"id": "date_year", "type": "text"},  # type downgrade
+                {"id": "datetime", "type": "timestamp"},
+                {"id": "duration", "type": "text"},  # type fallback
+                {"id": "geojson", "type": "json"},
+                {"id": "geopoint", "type": "text"},  # type fallback
+                {"id": "integer", "type": "int"},
+                {"id": "number", "type": "float"},
+                {"id": "object", "type": "json"},
+                {"id": "string", "type": "text"},
+                {"id": "time", "type": "time"},
+                {"id": "year", "type": "int"},
+                {"id": "yearmonth", "type": "text"},  # type fallback
```

Comment:
We're falling back `any`, `duration`, `geopoint` and `yearmonth` to `text` here. Do all of those seem right?